src/broker/messaging.bif

```diff
@@ -179,13 +189,25 @@ type Cluster::Pool: record;
 ##       to it.
 ##
 ## Returns: true if the message is sent.
-function Cluster::publish_rr%(pool: Pool, key: string, ...%): bool
+function Cluster::publish_rr%(pool: Pool, key: any, ...%): bool
```

Comment:
If you pass a non-string to a string argument, the function is still called, correct? Just the type is still wrong on the object? The reason I'm asking is that leaving this as a `string` gives more indication to a user what should be passed without having to read the documentation. If it has to be `any` for everything to work, that's fine too though.

Reply:
There's two things at play:
* bifcl makes this a `global Cluster::publish_rr: function(va_args: any): bool`, which is excluded from static type checking
* bifcl *does* generate the following preamble with an unconditional `->AsStringVal()`:
```
        zeek::Val* pool = (zeek::Val*) ((*BiF_ARGS)[0].get());
        zeek::StringVal* key = (zeek::StringVal*) ((*BiF_ARGS)[1].get()->AsStringVal());
```

So, today, in 5.2.1, you can trigger a crash as follows, which this patch makes a runtime error.
```
$ zeek -e 'Cluster::publish_rr(Cluster::Pool(), 1/tcp)'
fatal error in <no location>: Val::CONVERTER (port/string) (1/tcp)
Aborted (core dumped)
```

I agree that dropping the `: string` part is subpar. An alternative that I hadn't thought of before would be to teach `bifcl` not to do/assume the type conversions when dealing with those vargs bifs. Thoughts?

This all is treading around the bigger Zeek script vaargs issue though.

Reply:
> I agree that dropping the : string part is subpar. An alternative that I hadn't thought of before would be to teach bifcl not to do/assume the type conversions when dealing with those vargs bifs. Thoughts?

I wonder how much code would break if you did this though. Ugh.

I guess this is a good enough fix for now, but we should definitely flag it in the other issue so that it can be properly fixed if/when someone gets to that.